### DISCUSSION TITLE
Protein Grouping

### DISCUSSION THREAD

**Discussion by SauerTho** [vdemichev/DiaNN#107] [Q&A] [answered]
Hey,

I have a question regarding the protein grouping with respect to the reported protein groups. I have noticed that some proteins are reported uniquely and additionally are grouped together with other proteins, sometimes even multiple groups. Using alternive software tools, I have never noticed this. Could you give me some insight into this? Is there an option that allows to report a protein in just one group?

Kind regards
Thorben

**Comment by vdemichev** [comment, marked as answer]
Hi Thorben,

Please see my posts in this thread: https://github.com/vdemichev/DiaNN/issues/22
Also, in the recent DIA-NN there's an option "--relaxed-prot-inf", which makes it perform protein grouping similarly to FragPipe and Spectronaut.

Best wishes,

Vadim

**Reply by SauerTho** [reply]
Dear Vadim,
I have missed that one, thank you!

All the best,
Thorben

**Reply by JianAtSeer** [reply]
Dear Vadim:
I have a follow up question regarding the "--relazed-prot-inf" option, I understand this is doing the same greedy set cover algorithm as in other tools right?
My question is does the options "--pg_level" affect this step at all?  i.e. if i set it to gene level vs protein level, does it affect how the grouping is done? or it just affect the definition of proteotypic peptide and fdr scoring? Thanks


cheers


Jian

**Reply by vdemichev** [reply]
Hi Jian,

Yes, --pg-level affects protein inference in any of the modes. For example, if it is set to 'Genes' and a particular protein has no gene annotation, while other proteins do have, these other proteins will be prioritised for inclusion into the protein group.

Best,
Vadim

**Reply by JianAtSeer** [reply]
Hi Vadim:
Thanks for the quick reply.  I am still a little confused about the two different mode of protein inference and I was wondering if you can clarify.  1) Do both default mode and "--relaxed-prot-inf"  mode perform the greedy set cover algorithm to infer proteins from peptides? 2)and the difference between the two mode lies in how you reported the protein group at the end depending on whether you assigned the shared peptides uniquely to only one protein/group  (presumably the one with most peptide); 3) If the default mode is actually performing a different protein inference method, is there a reference or documentation that describe this method in more detail? Thanks a lot.


cheers


Jian

**Reply by vdemichev** [reply]
Hi Jian, yes both do greedy set cover. The difference is that if peptide X can originate from proteins A & B, peptide Y - from B & C, peptide Z - from A & C, then default method will report protein groups A;B for X, B;C for Y, A;C for Z, while --relaxed-prot-inf will report A for X and C for Y and Z.

Best,
Vadim
